cli/flox-rust-sdk/src/models/environment/core_environment.rs

```diff
@@ -937,15 +936,27 @@ impl EditResult {
             // todo: use a single toml crate (toml_edit already implements serde traits)
             // TODO: use different error variants, users _can_ fix errors in the _new_ manifest
             //       but they _can't_ fix errors in the _old_ manifest
-            let old_manifest: Manifest =
+            let old_manifest: TypedManifest =
                 toml::from_str(old_manifest).map_err(CoreEnvironmentError::DeserializeManifest)?;
-            let new_manifest: Manifest =
+            let new_manifest: TypedManifest =
                 toml::from_str(new_manifest).map_err(CoreEnvironmentError::DeserializeManifest)?;
-            // TODO: some modifications to `install` currently require re-activation
-            if old_manifest.hook != new_manifest.hook || old_manifest.vars != new_manifest.vars {
-                Ok(Self::ReActivateRequired { store_path })
-            } else {
-                Ok(Self::Success { store_path })
+
+            match (&old_manifest, &new_manifest) {
+                (TypedManifest::Pkgdb(old), TypedManifest::Pkgdb(new)) => {
+                    if old.hook != new.hook || old.vars != new.vars {
+                        Ok(Self::ReActivateRequired { store_path })
+                    } else {
+                        Ok(Self::Success { store_path })
+                    }
+                },
+                (TypedManifest::Catalog(old), TypedManifest::Catalog(new)) => {
+                    if old.hook != new.hook || old.vars != new.vars {
+                        Ok(Self::ReActivateRequired { store_path })
+                    } else {
+                        Ok(Self::Success { store_path })
+                    }
+                },
+                _ => Ok(Self::ReActivateRequired { store_path }),
```

Comment:
not quoite sure how we should handle this mixed manifest case, opinions welcome

Reply:
I think the transition from Pkgdb -> Catalog is a transition we may have to recommend people make (e.g. for, https://github.com/flox/flox/issues/1466, if we try to perform migration automatically and it fails, we'll have to say `please 'flox edit', set 'version = 1', and fix this error`. And I think the logic would essentially be the same as what it is for Catalog -> Catalog or Pkgdb -> Pkgdb.

I guess Catalog -> Pkgdb should be possible but is an edge case we don't really care about.

Reply:
Think we should add this too?

Reply:
I think we can look at that with #1466?
Otherwise, i'd not be completely sure what youre looking for here..

Reply:
We can do it in 1466. I was thinking something like, although I'd have to double check the `to_string` doesn't invalidate the comparison

```
                (TypedManifest::Pkgdb(old), TypedManifest::Catalog(new)) => {
                    if toml::to_string(&old.hook) != toml::to_string(&new.hook)
                        || toml::to_string(&old.vars) != toml::to_string(&new.vars)
                        || toml::to_string(&old.profile) != toml::to_string(&new.profile)
                    {
                        Ok(Self::ReActivateRequired { store_path })
                    } else {
                        Ok(Self::Success { store_path })
                    }
                },
                (TypedManifest::Catalog(old), TypedManifest::Pkgdb(new)) => {
                    if toml::to_string(&old.hook) != toml::to_string(&new.hook)
                        || toml::to_string(&old.vars) != toml::to_string(&new.vars)
                        || toml::to_string(&old.profile) != toml::to_string(&new.profile)
                    {
                        Ok(Self::ReActivateRequired { store_path })
                    } else {
                        Ok(Self::Success { store_path })
                    }
                },
```

Reply:
if they have the same order to string should work

Reply:
ok done, using toml::Value in hope that's less order dependent